r2/r2/lib/automoderator.py

```diff
@@ -98,6 +98,10 @@ def replace_placeholders(string, data, matches):
         "{{author}}": data["author"].name,
         "{{body}}": getattr(item, "body", ""),
         "{{subreddit}}": data["subreddit"].name,
+        "{{author_flair_text}}": getattr(data["author"],
```

Comment:
This isn't quite the right way to get their flair, since it won't take into account whether they've disabled their flair in the subreddit. I would put something like this before the `replacements = {` line above:

```
# get author flair text/class
sr_id = data["subreddit"]._id
author_flair_text = data["author"].flair_text(sr_id, obey_disabled=True)
author_flair_css_class = data["author"].flair_css_class(
    sr_id, obey_disabled=True)
```

And then here, just do (note that I'm also changing `{{author_flair_class}}` to `{{author_flair_css_class}}` for consistency with the names AutoMod uses for flair checks):

    "{{author_flair_text}}": author_flair_text,
    "{{author_flair_css_class}}": author_flair_css_class,

It's not necessary to worry about the case where the user doesn't have flair (or if they've disabled it in that subreddit), since `_force_unicode(None)` returns an empty string anyway, so it'll already work properly.